src/components/picker/new.tsx

```diff
@@ -0,0 +1,266 @@
+// TODO: deprecate all places where we check if _.isPlainObject
+// TODO: deprecate getItemValue prop
+// TODO: deprecate getItemLabel prop
+// TODO: Add initialValue prop
+// TODO: consider deprecating renderCustomModal prop
+// TODO: deprecate onShow cause it's already supported by passing it in pickerModalProps
```

Comment:
1) Why change `_.isPlainObject` usage?
2) `getItemValue` and `getItemLabel` deprecation started, so we can remove comments.
3) Why not start with the deprecation of `onShow`?

Reply:
1. For TS, when using comparison like this (`typeof value === 'object'`), Typescript actually interpret the right type of the `value` prop. When using lodash unfortunately it doesn't. 
2. They didn't really started, I just added the @deprecated annotation long time ago. We need to address it accordingly. 
3. That's not the purpose of this PR, notice that most of the TODOs here are old. 

The main purpose of this PR is migrate to TS and function component, which is big as it is, I didn't want to add more stuff.